pkg/dashboard/resource/project.go

```diff
@@ -134,23 +235,111 @@ func (pr *projectResource) Create(request *http.Request) (id string, attributes
 	return
 }
 
-// returns a list of custom routes for the resource
-func (pr *projectResource) GetCustomRoutes() ([]restful.CustomRoute, error) {
+func (pr *projectResource) importProject(projectImportInfoInstance *projectImportInfo) (id string,
+	attributes restful.Attributes, responseErr error) {
 
-	// since delete and update by default assume /resource/{id} and we want to get the id/namespace from the body
-	// we need to register custom routes
-	return []restful.CustomRoute{
-		{
-			Pattern:   "/",
-			Method:    http.MethodPut,
-			RouteFunc: pr.updateProject,
-		},
-		{
-			Pattern:   "/",
-			Method:    http.MethodDelete,
-			RouteFunc: pr.deleteProject,
-		},
-	}, nil
+	pr.Logger.DebugWith("Checking if project exists", "project", projectImportInfoInstance.Project.Meta.Name)
+
+	projects, err := pr.getPlatform().GetProjects(&platform.GetProjectsOptions{
+		Meta: *projectImportInfoInstance.Project.Meta,
+	})
+	if err != nil || len(projects) == 0 {
+		pr.Logger.DebugWith("Project doesn't exist, creating it", "project", projectImportInfoInstance.Project.Meta.Name)
+		err = pr.createAndWaitForProjectCreation(projectImportInfoInstance.Project)
+		if err != nil {
+			return "", nil, nuclio.WrapErrInternalServerError(err)
+		}
+	}
+
+	pr.Logger.Debug("Importing project functions")
+
+	var failedFunctions []restful.Attributes
+	for functionName, functionImport := range projectImportInfoInstance.Functions {
+		if functionImport.Function.Meta.Labels == nil {
+			functionImport.Function.Meta.Labels = map[string]string{}
+		}
+		functionImport.Function.Meta.Labels["nuclio.io/project-name"] = projectImportInfoInstance.Project.Meta.Name
+
+		err = pr.postFunctionForImport(functionImport.Function)
+		if err != nil {
+			pr.Logger.WarnWith("Failed posting function", "name", functionName, "err", err)
+			failedFunctions = append(failedFunctions, restful.Attributes{
+				"function": functionName,
+				"error":    err.Error(),
+			})
+		}
+	}
+
+	attributes = restful.Attributes{
+		"createdFunctionAmount": len(projectImportInfoInstance.Functions) - len(failedFunctions),
+		"failedFunctionsAmount": len(failedFunctions),
+		"failedFunctions":       failedFunctions,
+	}
+
+	return
+}
+
+func (pr *projectResource) postFunctionForImport(functionInfo *functionInfo) error {
+	jsonStr, err := json.Marshal(functionInfo)
+	if err != nil {
+		return err
+	}
+	urlStr := "http://localhost" + pr.getListenAddress() + "/api/functions"
+	request, err := http.NewRequest("POST", urlStr, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return err
+	}
+
+	request.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		return err
+	}
+	if response.StatusCode == http.StatusConflict {
+		return errors.New("function name already exists")
+	}
+	if response.StatusCode != http.StatusAccepted {
+		return errors.New("received unexpected status code: " + strconv.FormatInt(int64(response.StatusCode), 10))
```

Comment:
use `errors.Newf("blablabla: %d", 1)` for an instance

Reply:
no such function `Newf` in `nuclio/errors` pkg.
Will use `errors.New(fmt.Sprintf("blablabla: %d", 1))`